Home Work № 8

skyseedim opened the pull request.

skyseedim committed:
Home Work № 8

NicknameSoHard commented on HW8.py:
```diff
@@ -0,0 +1,340 @@
+        else:
+            print('Всё корректно!')
+
+Date.is_valid('10-01-2021')
+Date.is_valid('10-01-256549161')
+Date.is_valid('31-02-2020')
+Date.is_valid('31-13-2020')
+
+num = Date('28-02-1905')
+print('Число: ', num.date_number())
```
Еще для большего удобства я бы предложил дату отправлять в конструктор, а эти методы вызывать из него - или инициализировали объект, или вывалились с ошибкой, например. Тогда получится что у вас есть объект с точно валидной цифровой датой, которым вы можете дальше управлять. Будет поудобнее)

NicknameSoHard commented on HW8.py:
```diff
@@ -0,0 +1,340 @@
+            else:
+                num_transfer = input(f"Введите количество оборудования для отправки в подразделение {self.division}: ")
+
+        if id_num[:1] == 'P': #id поможет определить тип оборудования
+            stor_key = 'Printer'
+        elif id_num[:1] == 'S':
+            stor_key = 'Scanner'
+        elif id_num[:1] == 'C':
+            stor_key = 'Copier'
```
Как альтернативный вариант, вы можете использовать сам класс или какой-то из его методов чтобы получать ключ. Тогда единообразно все объекты сможете обрабатывать

NicknameSoHard commented on HW8.py:
```diff
@@ -0,0 +1,340 @@
+        self.real_part = compl_numb[0]
+        self.im_part = compl_numb[1]
+
+    def __add__(self, other):
+        return (self.real_part + other.real_part, self.im_part + other.im_part)
+
+    def __mul__(self, other):
+        return(self.real_part * other.real_part - self.im_part * other.im_part, self.real_part * other.im_part + \
+               other.real_part * self.im_part)
```
Обычно ведь операция возвращает результат того же типа что и слагаемые, но в виде нового объекта. Т.е. 2 + 2 = 4 (всё числа и всё разные объекты), а '2' + '2' = '22' (все строки и тоже разные объекты). Тогда получится что вы можете дальше оперировать и исходными данными, и их результатом. Так что здесь незачем изменять это правило и логичнее возвращать то же самое `Complex` в результате. Ведь все-равно перед `print` полученный объект будет преобразован в строчку в реализованном вами `__str__`.

NicknameSoHard commented on HW8.py:
```diff
@@ -0,0 +1,340 @@
+cmul = cnum1 * cnum2
+print(Complex(cmul))
+print(cnum1)
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
```
Убрали бы переносы бы, а то к чему они)

skyseedim committed:
Изменения в решениях задач 4-6, 7

Задача 4-6: Введён новый статический метод, определяющий id процедуры  поступления/передачи оборудования (Раньше id определялось каждый раз по ходу работы скрипта, теперь - через новый метод)
Задача 7:  Теперь переопределённые  операции сложение и умножение комплексных чисел возвращают Complex в результате для корректной работы print (так как __str__ переопределён)
Удалены лишние пустые строки
